roles/rhol_crc/tasks/main.yml

```diff
@@ -137,3 +137,17 @@
 - name: Add crc kubeconfig
   when: cifmw_rhol_crc_creds | bool
   ansible.builtin.import_tasks: add_crc_creds.yml
+
+- name: Configure DNS in CRC VM
```

Comment:
I see some issues calling this from rhols_crc/main.yml:
- there's a good chance the networks would still be CRC default ones in there, so the whole update would be for nothing.
- we won't get the updated dnsmasq config from within the reproducer: it deploys CRC using the defaults, then creates an overlay and then only connects to the networks - meaning we would end with a faulty CRC (yes, it's currently broken in there :( ).

So I propose to NOT call those tasks from within rhol_crc, but from the pelace where CRC is really configured/consumed:
- reproducer/configure_crc.yml
- (the playbook responsible to bootstrap nested CRC)

Any thoughts?

Reply:
To be honest I thought rhol_crc was the place crc is really configured/consumed as per the readme `Role to deploy get, configure and start RHOL/CRC instance`

We have 3 situations I'm trying to cover:
- Nested CRC covered by rhol_crc role as I don't see a playbook where we configure nested CRC, I just see us calling the role in `02-infra`
- reproducer CRC now covered by `configure_crc.yml`
- multinode CRC covered by `ci/playbooks/multinode-customizations.yml`

Reply:
right now I think we should be ok for the nested and multinode crc, but would fail for the reproducer crc case, since there is no `crc` node and the delegate below would fail

Reply:
For the reproducer case I added[1] which I don't think is defined when we call the rhol_crc role from reproducer so it will be skipped.
[1] 
```
  when:
    - cifmw_use_crc is defined
    - cifmw_use_crc | bool
```
Then in `roles/reproducer/tasks/configure_crc.yml` I use `delegate_to: crc-0`

How does this sound? It's difficult covering all the  scenarios, one more reason I wish we could drop nested CRC already (excluding reproducer :) )

Reply:
I think that should work, I'll duplicate the PR minus the zuul.d changes and test it downstream